tests/test_service_common.py

```diff
@@ -1939,23 +1939,23 @@ def test_checked_get(self):
             {},
         )
 
-        with self.assertRaisesRegex(exceptions.ValidationError,
-                                    "missing 'nonexistent'"):
+        with self.assertRaisesRegex(exceptions.BaseError,
+                                    "Missing required value"):
             common.checked_get(mapping, 'nonexistent', [], required=True)
 
-        with self.assertRaisesRegex(exceptions.ValidationError,
-                                    "missing 'nonexistent'"):
+        with self.assertRaisesRegex(exceptions.BaseError,
+                                    "Missing required value"):
             common.checked_get(mapping, 'nonexistent', {}, required=True)
 
         # bad value
         with self.assertRaisesRegex(
-                exceptions.ValidationError,
-                'invalid \'dict\', expected list, got: {"1337": 1337}',
+                exceptions.BaseError,
+                'Invalid type',
         ):
             common.checked_get(mapping, 'dict', [])
 
         with self.assertRaisesRegex(
-                exceptions.ValidationError,
-                r"invalid 'list', expected dict, got: \[1337\]",
+                exceptions.BaseError,
+                r"Invalid type",
```

Comment:
Can we extend this with a test that verifies that the output is helpful? For example, that it includes the name of the failing key, and preferably the invalid value too. (We should help the client by informing them _how_ they're wrong, rather than merely passing blame.)